src/adapter/adapter.js

```diff
@@ -0,0 +1,212 @@
+  },
+
+  /**  @returns {void} - Register listeners to this.player. */
+  registerListeners: function () {
+    var Event = this.player.Event
+    // References
+    this.references = []
+    this.references[Event.PLAY] = this.playListener.bind(this)
+    this.references[Event.LOAD_START] = this.playListener.bind(this)
+    this.references[Event.PAUSE] = this.pauseListener.bind(this)
+    this.references[Event.PLAYING] = this.playingListener.bind(this)
+    this.references[Event.ERROR] = this.errorListener.bind(this)
+    this.references[Event.SEEKING] = this.seekingListener.bind(this)
+    this.references[Event.SEEKED] = this.seekedListener.bind(this)
+    this.references[Event.PLAYER_STATE_CHANGED] = this.stateChangeListener.bind(this)
+    this.references[Event.ENDED] = this.endedListener.bind(this)
+    this.references[Event.CHANGE_SOURCE_STARTED] = this.forceEndedListener.bind(this)
+
+    // Register listeners
+    for (var key in this.references) {
+      this.player.addEventListener(key, this.references[key])
+    }
+  },
+
+  /**  @returns {void} - Unregister listeners to this.player. */
+  unregisterListeners: function () {
+
+    // unregister listeners
+    if (this.player && this.references) {
+      for (var key in this.references) {
+        this.player.removeEventListener(key, this.references[key])
+      }
+      this.references = []
+    }
+  },
+
+  bindLogger: function (logger) {
+    youbora.Log.error = logger.error.bind(logger)
+    youbora.Log.notice = logger.info.bind(logger)
+    youbora.Log.warn = logger.warn.bind(logger)
+    youbora.Log.debug = logger.debug.bind(logger)
+    youbora.Log.verbose = function () {
+    }
+  },
+
+  /** @returns {void} - Listener for 'play' event. */
+  playListener: function () {
+    if (!this.flags.isStarted) {
+      this.fireStart()
+      this.plugin.options['content.isLive.noSeek'] = !this.player.isDvr()
+      this.initialPlayhead = this.getPlayhead()
+    }
+  },
+
+  /** @returns {void} - Listener for 'timeupdate' event. */
+  timeupdateListener: function () {
+    this.fireStart()
+    this.fireJoin()
+  },
+
+  /** @returns {void} - Listener for 'pause' event. */
+  pauseListener: function () {
+    this.firePause()
+  },
+
+  /** @returns {void} - Listener for 'playing' event. */
+  playingListener: function () {
+    this.fireResume()
+    this.fireSeekEnd()
+    this.fireBufferEnd()
+    this.fireStart()
+    this.fireJoin()
+  },
+
+  /** @returns {void}
+   * @param {Object} e - object with payload including severity, code and data.
+   * - The name of the plugin.- Listener for 'error' event. */
+  errorListener: function (e) {
+    if (e.payload.severity === Error.Severity.CRITICAL) {
+      this.fireError(e.payload.code, e.payload.data)
+      this.fireStop()
+    }
+  },
+
+  /** @returns {void} - Listener for 'seeking' event. */
+  seekingListener: function () {
+    if (!this.flags.isBuffering &&
+      (this.initialPlayhead !== this.getPlayhead())) {
```

Comment:
Because a buffer can be converted into a see when after a fireBufferBegin() we call to fireSeekEnd(), so we don't need to call to a begin method again.
The second one is to workaround an issue: vod sends a seek just after start.